src/lib.ts

```diff
@@ -124,58 +125,79 @@ export function extractSchemaAndDoc(
         // `@gqlQueryField` and friends.
         .map((doc) => addImplicitRootTypes(doc))
         // Merge any `extend` definitions into their base definitions.
-        .map((doc) => mergeExtensions(doc))
+        .map((doc) => mergeExtensions(ctx, doc))
         // Perform custom validations that reimplement spec validation rules
         // with more tailored error messages.
         .andThen((doc) => customSpecValidations(doc))
```

Comment:
I've noticed that this will fail with external types since it notices the type has no fields. Can we move this below `addImportedSchemas`?

Reply:
Hmm, actually even disabling this seems to trigger validation errors on line 135 saying the external type has no fields.